### PR TITLE
Set wifi MAC address based on the contents of the EEPROM

### PR DESCRIPTION
**Pull request by willnewton**
This is an attempt to fix https://github.com/resin-os/resinos/issues/133

The MAC address for the wifi interface is stored in the EEPROM and needs to be set at boot, otherwise it remains DE:AD:BE:EF:00:00

### CODE REVIEW

File: layers/meta-resin-beaglebone/recipes-core/images/resin-image.bbappend [hunk 1]
```diff
@@ -1,3 +1,3 @@
 include resin-image.inc
 
-IMAGE_INSTALL_append_beaglebone = " bb-org-overlays fix-mmc-bbb"
+IMAGE_INSTALL_append_beaglebone = " bb-org-overlays fix-mmc-bbb bb-wl18xx-wlan0"
```
Comment (agherzan) [inline layers/meta-resin-beaglebone/recipes-core/images/resin-image.bbappend]: I would have split this into another PR. Just a NIT.

File: layers/meta-resin-beaglebone/recipes-connectivity/bb-wl18xx-wlan0/files/bb-wl18xx-wlan0.service [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=BB WL18xx wlan0 Service
+Before=network-pre.target
+Wants=network-pre.target
+After=local-fs.target
+
+[Service]
+ExecStart=/usr/bin/bb-wl18xx-wlan0
+Type=oneshot
```
Comment (agherzan) [inline layers/meta-resin-beaglebone/recipes-connectivity/bb-wl18xx-wlan0/files/bb-wl18xx-wlan0.service]: NIT: Type first.

File: layers/meta-resin-beaglebone/recipes-connectivity/bb-wl18xx-wlan0/files/bb-wl18xx-wlan0 [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+# Modified by willn@resin.io. Original at:
```
Comment (agherzan) [inline layers/meta-resin-beaglebone/recipes-connectivity/bb-wl18xx-wlan0/files/bb-wl18xx-wlan0]: Would be nice to state the modifications done so we can avoid a manual diff to investigate it.

File: layers/meta-resin-beaglebone/recipes-connectivity/bb-wl18xx-wlan0/bb-wl18xx-wlan0.bb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+DESCRIPTION = "Set wifi MAC address"
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://${RESIN_COREBASE}/COPYING.Apache-2.0;md5=89aea4e17d99a7cacdbeed46a0096b10"
+
+SRC_URI = " \
+	file://bb-wl18xx-wlan0 \
+	file://bb-wl18xx-wlan0.service \
+"
+
+inherit systemd
```
Comment (agherzan) [inline layers/meta-resin-beaglebone/recipes-connectivity/bb-wl18xx-wlan0/bb-wl18xx-wlan0.bb]: We should inherit allarch too.
As well this should include a COMPATIBLE_MACHINE .

**Comment by willnewton** [comment]
Hopefully the above issues should now be resolved.
